RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -132,12 +132,35 @@ class PumpOpsSynchronous {
             throw lastError
         }
     }
+    
+    private func pumpResponding() -> Bool {
+        do {
+            let msg = makePumpMessage(.GetPumpModel)
+            let response = try sendAndListen(msg, retryCount: 1)
+            
+            if response.messageType == .GetPumpModel && (response.messageBody as? GetPumpModelCarelinkMessageBody) != nil {
+                return true
+            }
+        } catch {
+        }
+        return false
+    }
+    
+
 
     private func wakeup(duration: NSTimeInterval = NSTimeInterval(minutes: 1)) throws {
         guard !pump.isAwake else {
             return
         }
         
+        if pumpResponding() {
+            NSLog("Pump responding despite our wake timer having expired. Extending timer")
+            // By my observations, the pump stays awake > 1 minute past last comms. Usually
```

Comment:
Yes, it certainly does.  I saw evidence of this even before hooking up the power monitor.  Just never measured it and took advantage of it in code.